src/Target.h

```diff
@@ -258,6 +258,16 @@ struct Target {
     /** Was libHalide compiled with support for this target? */
     EXPORT bool supported() const;
 
+    /** Given a size (in bytes), return True if the allocation size can fit
```

Comment:
Good points. The problem is that the profiler is a lowering pass, and needs
to understand at that point whether or not something is going to end up on
the stack so that it can simulate stack usage to determine worst-case peak
stack. CodeGen_C also had this repeated logic (and it was inconsistent). So
this seemed like target-dependent logic that is now used in three different
places. If it goes in CodeGen then we've broken the wall that previously
separated lowering passes from LLVM-using stuff.

On Tue, Mar 22, 2016 at 1:24 PM, Zalman Stern <notifications@github.com>
wrote:

> In src/Target.h
> <https://github.com/halide/Halide/pull/1078#discussion_r57062236>:
>
> > @@ -258,6 +258,16 @@ struct Target {
> >      /** Was libHalide compiled with support for this target? */
> >      EXPORT bool supported() const;
> >
> > +    /** Given a size (in bytes), return True if the allocation size can fit
>
> I should add that one reason I am sensitive to this is because Target.h
> sits on the boundary between being in the compiler and in the runtime. I.e.
> there are cases where one wants to be able to parse and manipulate target
> strings in the runtime. The more of this type of functionality we roll into
> it, the further we get from being able to do that.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/1078/files/4eb798e03cc76a9fc943da2a77e52998200eb4ba#r57062236>
>